architecture/protocol_config.go

```diff
@@ -0,0 +1,134 @@
+package architecture
+
+type CommandParseOptions struct {
+	ExpectedLength int
+	WaitingForMore bool
+	Params         []string
+	Name           CommandName
+}
+
+type CommandReplyOptions struct {
+	Result   bool
+	Message  string
+	Param    string
+	UseJobID bool
+}
+
+var cmdParseOptions map[CommandName]CommandParseOptions
+var cmdReplyOptions map[CommandName]CommandReplyOptions
+
+func init() {
+	cmdParseOptions = map[CommandName]CommandParseOptions{
+		USE: CommandParseOptions{
```

Comment:
Please remove `CommandParseOptions` here as the type is redundant. It would look much better without the repeating types I think :)